pkg/ast/pipesearch/searchQueryParser.go

```diff
@@ -138,6 +138,18 @@ func createMatchAll(qid uint64) *ASTNode {
 	return rootNode
 }
 
+func updatePositionForGenEvents(aggs *QueryAggregators) {
+	node := aggs
+	position := 1
+	for node != nil {
+		if node.GenerateEvent != nil {
+			node.GenerateEvent.EventPosition = position
+			position++
+		}
+		node = node.Next
+	}
+}
```

Comment:
This function is required to assign the generate events an order number i.e. the position of generate event in the aggs list. This is required for creating keys. I think using a specified position/order from start to end of the list is better than using a random number.

Reply:
>  I think using a specified position/order from start to end of the list is better than using a random number.

I agree. Are you saying that if we set the `EventPosition` while building the `QueryAggregators` with `searchPipeCommandsToASTnode()` then we'd have to set the value to a random number on each element instead of use consecutive numbers? Don't we build each `QueryAggregator` in order?